sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -0,0 +1,148 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import uuid
+from base64 import b64decode
+from datetime import datetime
+from azure.core._utils import _convert_to_isoformat, TZ_UTC
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Dict
+
+
+__all__ = ["CloudEvent"]
+
+
+class CloudEvent(object):  # pylint:disable=too-many-instance-attributes
+    """Properties of the CloudEvent 1.0 Schema.
+    All required parameters must be populated in order to send to Azure.
+    If data is of binary type, data_base64 can be used alternatively. Note that data and data_base64
+    cannot be present at the same time.
+    :param source: Required. Identifies the context in which an event happened. The combination of id and source must
+     be unique for each distinct event. If publishing to a domain topic, source must be the domain name.
+    :type source: str
+    :param type: Required. Type of event related to the originating occurrence.
+    :type type: str
+    :keyword data: Optional. Event data specific to the event type. If data is of bytes type, it will be sent
```

Comment:
This comment applies to the event grid sender client, right?

Reply:
it can technically be any request
"will be sent as data_base64 in the outgoing request." is the wording here.

Reply:
But there is no code here that will do base64 encoding of the data in this type? I would have understood it better if there was a `to_dict` method that would convert `self.data` to `'data_base64': <base 64 encoded self.data>`.

In other words, are you trying to document a requirement for the consumer of this data here?

Reply:
i'm trying to docuement the implementation detail of json serialization